copy-this-template-directory/index.md

```diff
@@ -1,16 +1,924 @@
+
+`FLOWS_ENV_ALLOW_LIST=GCALENDARHOOKSECRET,GCALENDARHOOKURL`
+
+&nbsp; 
+
+**Now a Flows.**
+
+There are 4 Flows. For each Flow I'll show an overview picture and then each node screen with config also as text, so you can copy-paste it.
+
+&nbsp; 
+
+### Flow "Google Calendar event Proxy"
+
+As I mentioned - it's not very straightforward to organize a stream of updates from Google Calendar with Google Apps Script.
+We will use [push notifications](https://developers.google.com/calendar/api/guides/push) that will call our published Google Apps Script when there are changes in the calendar. The catch though, these push notifications provide data within Header but for incoming requests, Google Apps Script can't read headers, only the body. That's why we will use Directus Flow as a simple proxy, it's webhook URL will be registered as the address for push notifications, it will receive data, save the header as a body, and send it to our published Google Apps Script.
+
+![whole flow](/copy-this-template-directory/directus_flow_1_full_.png "whole flow")
+
+***
+
+- Trigger node - webhook:
+
+![trigger node](/copy-this-template-directory/directus_flow_1_01_.png "trigger node")
+
+
+***
+
+- Node 2 - "Webhook / Request URL"
+
+![request node](/copy-this-template-directory/directus_flow_1_02_.png "request node")
+
+URL is `{{$env.GCALENDARHOOKURL}}` - the actual value in the environment variable will be set after Google Apps Script is published.
+
+Request body:
+```js
+{
+  "headers": {{$trigger.headers}}
+}
+```
+
+> note that {{$trigger.headers}} is not quoted!
+
+After you save this Flow, copy the resulting webhook URL somewhere (the URL can be copied from the trigger node, when in view mode, not in edit mode).
+
+***
+
+&nbsp; 
+
+### Flow "Send delete event to Google Calendar"
+
+Processing of the Delete event is a bit different from processing of Create / Update (next flow). It should be set to blocking, cause we need to "intercept" the delete command and read item data - we need to know the id of the Google Calendar event, so we can send it to Published Google Apps Script.
+
+![whole flow](/copy-this-template-directory/directus_flow_2_full_.png "whole flow")
+
+***
+
+- Trigger node - Event Hook
+
+![trigger node](/copy-this-template-directory/directus_flow_2_01_.png "trigger node")
+
+make sure that you have the same config:
+
+Type - Filter(Blocking)
+
+Scope - items.delete
+
+Collections - a collection of your choice
+
+Response - data of last operation
+
+
+***
+
+- Node 2 - "Read data"
+
+![node 02](/copy-this-template-directory/directus_flow_2_02_.png "node 02")
+
+IDs edit raw value to:
+```js
+[
+    "{{$trigger.payload[0]}}"
+]
+```
+Query is empty
+
+
+***
+
+- Node 3 - "Webhook / Request URL"
+
+![node 03](/copy-this-template-directory/directus_flow_2_03_.png "node 03")
+
+URL is `{{$env.GCALENDARHOOKURL}}` - the actual value in the environment variable will be set after Google Apps Script is published.
```

Comment:
We don't need to specify the value is coming later as we have already done that in this tutorial.